hack/test.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -e
+
+COVERAGE_MINIMUM=20
+COVERAGE_PASS="true"
+
+coverage run -m unittest discover src
+coverage report -m | tee coverage.log
+
+for FILE in $(find src/ -type f | grep -v -e "__" -e request_helper -e "/test" -e "src/[^/]*.py" | grep py$);
+do
+    if [ $(grep "$FILE" coverage.log | wc -l) -gt 0 ];
+    then
+        COVERAGE=$(grep "$FILE" coverage.log | awk '{print $4}' | sed 's/%//g')
```

Comment:
This is me being anal again, so feel free to ignore, but this can be consolidated from five subprocesses to one:

```suggestion
    COVERAGE=$(awk '$1 == "'$FILE'" {sub("%", ""); print $4}' coverage.log)
    if [ "$COVERAGE" ]
    then
```

Reply:
My use of `awk` is basically a limited "break on this char and print some stuff".. some day I should get better use out of it!